src/object.rs

```diff
@@ -346,6 +346,47 @@ pub trait ObjectFactory: Debug + Send + Sync {
         )
     }
 
+    fn default_object_derive(
+        &self,
+        template: &[CK_ATTRIBUTE],
+        origin: &Object,
+    ) -> KResult<Object> {
+        /* FIXME: handle CKA_DERIVE_TEMPLATE */
+
+        let mut obj = self.internal_object_create(
+            template,
+            OAFlags::UnsettableOnUnwrap,
```

Comment:
I see that unwrapping and deriving operations have some similar characteristics but reusing the "upwrap" here might sounds confusing so it should get at least a comment of we can not figure out a common name for these two operations and name the flag better `UnsettableOnUnwrapOrDerive`?

Reply:
Ah yeah ... I'll think about this, maybe there is another more generic term that can subsume both behaviors somehow.
Naming ... the easy part of programming :)

Reply:
Although not exactly part of this PR this comment made me look more carefully at the flags and I realized there was a lot of inconsistency and the code had changed at some point and stopped enforcing some stuff it was supposed to.

I attached an extra patch to deal with that.